tools/ClassGen/MemberType.cpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "MemberType.h"
+
+#define DEFINE_BASIC_MEMBER_TYPE(memberType, returnTypename, storageTypename,  \
+                                 ctorArgTypeName)                              \
+  MemberTypeInfo k##memberType##TypeInfo{MemberType::memberType,               \
+                                         returnTypename, storageTypename,      \
+                                         ctorArgTypeName};
+
+DEFINE_BASIC_MEMBER_TYPE(TypeRef, "TypeRef", "TypeRef", "TypeRef")
+DEFINE_BASIC_MEMBER_TYPE(Float, "float", "float", "float")
+DEFINE_BASIC_MEMBER_TYPE(Unsigned, "unsigned_t", "unsigned_t", "unsigned_t")
+DEFINE_BASIC_MEMBER_TYPE(Boolean, "bool", "bool", "bool")
+DEFINE_BASIC_MEMBER_TYPE(String, "std::string", "std::string", "std::string")
+DEFINE_BASIC_MEMBER_TYPE(VectorFloat, "llvm::ArrayRef<float>",
+                         "std::vector<float>", "std::vector<float>")
+DEFINE_BASIC_MEMBER_TYPE(VectorUnsigned, "llvm::ArrayRef<unsigned_t>",
+                         "std::vector<unsigned_t>", "std::vector<unsigned_t>")
+DEFINE_BASIC_MEMBER_TYPE(VectorSizeT, "llvm::ArrayRef<size_t>",
+                         "std::vector<size_t>", "std::vector<size_t>")
+DEFINE_BASIC_MEMBER_TYPE(VectorNodeValue, "NodeValueArrayRef",
+                         "std::vector<NodeHandle>", "std::vector<NodeValue>")
+
```

Comment:
Looks like you do not win a lot by using macro definitions here. You could just as well define these types directly.